### DISCUSSION TITLE
Jepsen, Hermitage, etc. testing?

### DISCUSSION THREAD

**Discussion by dumblob** [juicedata/juicefs#2544] [General]
**What would you like to be added**:

Jepsen, Hermitage, etc. test results for every (incl. minor) release (ideally automated by CI).

**Why is this needed**:

Jepsen tests are the very minimum ("entry-level") in the today's DB world. Without Jepsen a DB "does not exist" (seen through the "production SW" lenses).

**Comment by davies** [comment]
The Jepsen test should be run against the DB directly, not against JuiceFS, which is just an application on top of the DB.

For JuiceFS, we already have a lots of file system test suite.

**Reply by dumblob** [reply]
@davies thank you for quickly chiming in. I am certain this is a huge misunderstanding.

Jepsen is an **independent** test suite for any end-user-facing distributed system **storing data**. If JuiceFS wants to become trustworthy an independent **recognized** testing is a must.

Therefore I am also not sure why this **issue** got converted to a discussion. Could you convert it back and link these new comments in the newly (re-)created issue? Thank you!

**Reply by davies** [reply]
> Jepsen is an **independent** test suite for any end-user-facing distributed system **storing data**. If JuiceFS wants to become trustworthy an independent **recognized** testing is a must.

Jepsen can be used to test any distributed system, the published test results are mostly transactional databases, I can't find any distributed file system has conducted a Jepsen test. So we are not sure Jepsen is the right tool to test distributed file system in practice. All the well-known file systems (HDFS, LustreFS, GPFS, MooseFS) have not done the Jepsen test, so we think it's not a MUST.
 
> Therefore I am also not sure why this **issue** got converted to a discussion. Could you convert it back and link these new comments in the newly (re-)created issue? Thank you!

Currently, we use issue as the TODO list, and I'm not sure how to do a Jepsen test against JuiceFS, so convert it into discussion. Once we can figure out what's the things to do, then we can move it back to issues.
